src/NServiceBus.Extensions.DependencyInjection.Lamar.AcceptanceTests/NServiceBus.Extensions.DependencyInjection.Lamar.AcceptanceTests.csproj

```diff
@@ -1,17 +1,23 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net461;netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp3.1</TargetFrameworks>
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\NServiceBus.Extensions.DependencyInjection\NServiceBus.Extensions.DependencyInjection.csproj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.0" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net461'">
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.1.0" />
```

Comment:
Is this because of Lamar's weird version requirements?

Reply:
Ugh, that package is totally broken and should not be be doing that.

Reply:
Ugh, this is a nasty thing to have to deal with. The single package is trying to support ASP.NET Core 2.x and 3.x, splitting by TFM.

Having different versions of dependencies based on TFM is not a great thing to do.

Reply:
Ah, the package is an "ASP.NET Core" DI package, and has references to an ASP.NET Core package as well. I think that's the crux of the problem. It's not just pure Microsoft.Extensions.

Reply:
yeah the package is a bit weird but I think there is currently no real workaround to targeting specific TFM dependencies? It's mostly due to multi-targeting our tests, for users, this shouldn't be an issue?